spec/node_spec.rb

```diff
@@ -65,20 +65,19 @@ module Ferrum
       end
     end
 
-    context "when the element is not in the viewport of parent element", skip: true do
+    context "when the element is not in the viewport of parent element" do
       before do
         browser.go_to("/ferrum/scroll")
       end
 
       it "scrolls into view" do
-        # FIXME:
-        browser.at_xpath("//a[text() = 'Link outside viewport']").click
-        expect(browser.current_url).to eq("/")
+        browser.at_xpath("//a[text() = 'Link outside viewport']").click(mode: :force)
+        expect(browser.current_url).to eq(base_url("/"))
       end
 
       it "scrolls into view if scrollIntoViewIfNeeded fails" do
```

Comment:
@route Those tests have references into `scrolls` is it based on?

> Indeed, we can't click things that are not on screen, so we try to scroll them in. If website has a custom logic around scrolling, this might break.

https://github.com/puppeteer/puppeteer/issues/2190#issuecomment-380254352

we probably may change tests descriptions due to this approach:

> I think it might be worth noting in the documentation that all of the click/mouse functionality depends on emulating real clicks where the link needs to be visible.

https://github.com/puppeteer/puppeteer/issues/2190#issuecomment-380262695

Reply:
Didn't get what you mean

Reply:
I was mean: the descriptions of the skipped tests are included scroll-references, like: `scrolls into view ...`, so, as I understand, the fix of those tests should contain some of `real clicks` implementation or we can use the emulating with avoid the scroll and, probably, with the change of the descriptions.

This is the main sentence which I was needed to choose the right way:

> This one is synthetic event and has nothing in common with real clicks

so, `synthetic event` about
> emulating real clicks where the link needs to be visible.

real clicks about:
scroll into element (somehow) to click on it